sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/schema/primitive/AvroNullSchema.java

```diff
@@ -0,0 +1,48 @@
+package com.azure.storage.internal.avro.implementation.schema.primitive;
+
+import com.azure.storage.internal.avro.implementation.AvroParserState;
+import com.azure.storage.internal.avro.implementation.schema.AvroSchema;
+
+import java.util.function.Consumer;
+
+/**
+ * A null is written as zero bytes.
+ */
+public class AvroNullSchema extends AvroSchema<AvroNullSchema.Null> {
+
+    /**
+     * Constructs a new AvroNullSchema.
+     *
+     * @param state The state of the parser.
+     * @param onResult The result handler.
+     */
+    public AvroNullSchema(AvroParserState state, Consumer<Null> onResult) {
+        super(state, onResult);
+    }
+
+    @Override
+    public void add() {
+        this.state.push(this);
+    }
+
+    @Override
+    public void progress() {
+        /* Just return a custom null object, then we're done. */
+        this.result = new Null();
+        this.done = true;
+    }
+
+    /**
+     * Can always make progress since null is zero bytes.
+     */
+    @Override
+    public boolean canProgress() {
+        return true;
+    }
+
+    /* We use a custom type to return null since null cannot be emitted in a Flux.
+       Users of the AvroParser must transform all NullSchema.Null objects to null if necessary. */
+    public static class Null {
+    }
```

Comment:
So we expect customer to see this type. For change feed that probably doesn't matter. but for QQ customer's are going to see that, so this should go to some model (or there should be mapping from this to some model when it comes to QQ implementation). What do you think?

Reply:
Since the Parser is internal, we can handle the conversion of this custom type to real null. I'll do that in both CF and QQ

Reply:
This essentially means I don't expect a customer to see this